### ISSUE TITLE
PostgreSQL support for "PARTITION BY" 

### ISSUE THREAD

**Issue by jdsada30** [sqlfluff/sqlfluff#1095]
-- If this is a parsing or linting issue, please include a minimal SQL example which reproduces the issue, along with the `sqlfluff parse` output, `sqlfluff lint` output and `sqlfluff fix` output when relevant. Finally, please add any labels that are related to the issue before submitting so that it can be addressed easily! --

## Expected Behaviour
I would expect statements with "PARTITION BY"  to be parsed by sqlfluff, as they are supported by PostgreSQL.

## Observed Behaviour
>== [test.sql] FAIL
L:   6 | P:   2 |  PRS | Found unparsable section: ' PARTITION BY RANGE
                       | (logdate)'

## Steps to Reproduce
Create a sql file with the following query:
>CREATE TABLE measurement (
    city_id int NOT NULL,
    logdate date NOT NULL,
    peaktemp int,
    unitsales int
) PARTITION BY RANGE (logdate);

Lint the file with the following command(s) :
>sqlfluff lint test.sql --dialect postgres or sqlfluff lint test.sql
## Version
Include the output of `sqlfluff --version` along with your Python version
I am using sqlfluff version 0.5.6 and python version 3.8.5.

This example comes from the official PostgreSQL [documentation](https://www.postgresql.org/docs/10/ddl-partitioning.html). I found similar results when executing queries from Github Project [vertx-jooq](https://github.com/jklingsporn/vertx-jooq).

**Comment by pwildenhain** [comment]
There is support for the `partition by` clause in the `SELECT` statement at least

`test_partition.sql`
```sql
select max(some_field) over (partition by some_other_field) from mytable
```

```
$ sqlfluff parse test_partition.sql
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        select_statement:
[L:  1, P:  1]      |            select_clause:
[L:  1, P:  1]      |                keyword:                                      'select'
[L:  1, P:  7]      |                [META] indent:
[L:  1, P:  7]      |                whitespace:                                   ' '
[L:  1, P:  8]      |                select_clause_element:
[L:  1, P:  8]      |                    function:
[L:  1, P:  8]      |                        function_name:
[L:  1, P:  8]      |                            function_name_identifier:         'max'
[L:  1, P: 11]      |                        bracketed:
[L:  1, P: 11]      |                            start_bracket:                    '('
[L:  1, P: 12]      |                            [META] indent:
[L:  1, P: 12]      |                            expression:
[L:  1, P: 12]      |                                column_reference:
[L:  1, P: 12]      |                                    identifier:               'some_field'
[L:  1, P: 22]      |                            [META] dedent:
[L:  1, P: 22]      |                            end_bracket:                      ')'
[L:  1, P: 23]      |                        whitespace:                           ' '
[L:  1, P: 24]      |                        over_clause:
[L:  1, P: 24]      |                            keyword:                          'over'
[L:  1, P: 28]      |                            whitespace:                       ' '
[L:  1, P: 29]      |                            bracketed:
[L:  1, P: 29]      |                                start_bracket:                '('
[L:  1, P: 30]      |                                [META] indent:
[L:  1, P: 30]      |                                window_specification:
[L:  1, P: 30]      |                                    partitionby_clause:
[L:  1, P: 30]      |                                        keyword:              'partition'
[L:  1, P: 39]      |                                        whitespace:           ' '
[L:  1, P: 40]      |                                        keyword:              'by'
[L:  1, P: 42]      |                                        [META] indent:
[L:  1, P: 42]      |                                        whitespace:           ' '
[L:  1, P: 43]      |                                        expression:
[L:  1, P: 43]      |                                            column_reference:
[L:  1, P: 43]      |                                                identifier:   'some_other_field'
[L:  1, P: 59]      |                                        [META] dedent:
[L:  1, P: 59]      |                                [META] dedent:
[L:  1, P: 59]      |                                end_bracket:                  ')'
[L:  1, P: 60]      |            whitespace:                                       ' '
[L:  1, P: 61]      |            [META] dedent:
[L:  1, P: 61]      |            from_clause:
[L:  1, P: 61]      |                keyword:                                      'from'
[L:  1, P: 65]      |                whitespace:                                   ' '
[L:  1, P: 66]      |                from_expression:
[L:  1, P: 66]      |                    [META] indent:
[L:  1, P: 66]      |                    from_expression_element:
[L:  1, P: 66]      |                        table_expression:
[L:  1, P: 66]      |                            table_reference:
[L:  1, P: 66]      |                                identifier:                   'mytable'
[L:  1, P: 73]      |                    [META] dedent:
[L:  1, P: 73]      |    newline:                                                  '\n'
```

Not sure why there isn't anything similar in the `CREATE` statement as well.
